tests/testthat/test-create_start_impute.R

```diff
@@ -105,4 +105,38 @@ describe("create_start_impute", {
                    dplyr::pull(impute),
                  "start_conc0")
   })
+
+  it("if drug column is not present, assumes is the same as analyte for imputation", {
+    # No drug but there is analyte
+    mydata_with_analyte <- mydata
+    mydata_with_analyte$dose$data$DRUG <- NULL
+    result_with_analyte <- create_start_impute(mydata_with_analyte)
+    expect_equal(result_with_analyte$intervals %>%
+                   dplyr::filter(USUBJID == 6, DOSNO == 1) %>%
+                   dplyr::pull(impute),
+                 "start_logslope")
+
+    # No drug and no analyte
+    mydata_no_analyte <- mydata
+    mydata_no_analyte$dose$data$DRUG <- NULL
+    mydata_no_analyte$conc$data$param <- NULL
+    result_no_analyte <- create_start_impute(mydata_no_analyte)
+    # For last subject now analyte and drug are matching, should be start_logslope
+    expect_equal(result_no_analyte$intervals %>%
+                   dplyr::filter(USUBJID == 6, DOSNO == 1) %>%
+                   dplyr::pull(impute),
+                 "start_logslope")
```

Comment:
nitpick: Since there is some logic to extract the result, I would assign it to a variable instead of directly inside `expect_equal` call.
```suggestion
    result_with_analyte_impute <- result_with_analyte$intervals %>%
                   dplyr::filter(USUBJID == 6, DOSNO == 1) %>%
                   dplyr::pull(impute)
    expect_equal(result_with_analyte_impute, "start_logslope")

    # No drug and no analyte
    mydata_no_analyte <- mydata
    mydata_no_analyte$dose$data$DRUG <- NULL
    mydata_no_analyte$conc$data$param <- NULL
    result_no_analyte <- create_start_impute(mydata_no_analyte)
    # For last subject now analyte and drug are matching, should be start_logslope
    result_no_analyte_impute <- result_no_analyte$intervals %>%
                   dplyr::filter(USUBJID == 6, DOSNO == 1) %>%
                   dplyr::pull(impute)
    expect_equal(result_no_analyte_impute, "start_logslope")
```